prow/plugins/config.go

```diff
@@ -802,11 +809,40 @@ func (c *Configuration) DcoFor(org, repo string) *Dco {
 	return &Dco{}
 }
 
+func oldToNewPlugins(oldPlugins map[string][]string) map[string]OrgPlugins {
+	newPlugins := make(map[string]OrgPlugins)
+	for repo, plugins := range oldPlugins {
+		newPlugins[repo] = OrgPlugins{
+			Plugins: plugins,
+		}
+	}
+	return newPlugins
+}
+
+func ToRawMessage(plugins interface{}) json.RawMessage {
+	raw, _ := yaml.Marshal(plugins)
+	return raw
+}
+
+func (c *Configuration) Plugins() map[string]OrgPlugins {
+	var oldPlugins map[string][]string
+	if err := yaml.Unmarshal(c.PluginsRaw, &oldPlugins); err != nil {
+		var newPlugins map[string]OrgPlugins
+		if err := yaml.Unmarshal(c.PluginsRaw, &newPlugins); err != nil {
+			logrus.WithError(err).Error("error unmarshalling Plugins config to oldPlugins and newPlugins formats")
```

Comment:
I see, but then what do I do with this error in the caller side?

Reply:
I think checkconfig need to catch this error and not letting the change go through. And for hook I would prefer to let it crash, with an error message asking for running checkconfig

Reply:
This custom unmarshaling is just for the transition period, to let people migrate from old to new format... otherwise such error is thrown when the yaml file is parsed, so maybe I should validate at that very beginning and not every time I need to access plugins?

Reply:
Ok, I will do like `ConfigUpdater` and redefine my own unmarshall method...